### PR TITLE
Disabled course list button

### LINKED ISSUE #28: Disable course list button for logged out users
The "Meine Kurse" button is still active for not logged-in users. It should be disabled.

### PR DESCRIPTION
**Pull request by Christoph309**
Fixed #28

- Set state of `course list button` to `disabled` when user is not logged in
- Set state of `course list button` to `disabled` when not `isUpperGrade`

### CODE REVIEW

File: app/src/main/java/com/rmkrings/fragments/preferences/GeneralPreferencesFragment.java [hunk 1]
```diff
@@ -190,13 +192,15 @@ private boolean isLowerGradeSelected(int val) {
      * @param forSelectedGradeVal - Currently selected grade index value.
      */
     private void setElementStates(int forSelectedGradeVal) {
+        System.out.println("?????????????????????????????????????????????" + forSelectedGradeVal);
```
Comment (Th3Ph4nt0m) [inline app/src/main/java/com/rmkrings/fragments/preferences/GeneralPreferencesFragment.java]: Debugging message should be removed here

File: app/src/main/java/com/rmkrings/fragments/preferences/GeneralPreferencesFragment.java [hunk 2]
```diff
@@ -84,14 +85,15 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
...
-        boolean isAuthenticated = AppDefaults.isAuthenticated();
+        isAuthenticated = AppDefaults.isAuthenticated();
         mUserName.setEnabled(!isAuthenticated);
         mPassword.setEnabled(!isAuthenticated);
 
         mClassPicker.setValue(AppDefaults.getSelectedClassRow());
         mGradePicker.setValue(AppDefaults.getSelectedGradeRow());
 
         setElementStates(AppDefaults.getSelectedGradeRow());
+        System.out.println(isAuthenticated + "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!");
```
Comment (Th3Ph4nt0m) [inline app/src/main/java/com/rmkrings/fragments/preferences/GeneralPreferencesFragment.java]: Debugging message should be removed here

**Review by Th3Ph4nt0m — CHANGES_REQUESTED** [review]

**Comment by Th3Ph4nt0m** [comment]
The state is not updated when switching from `Q2` to `keine`

**Review by Th3Ph4nt0m — APPROVED** [review]
Looks good for me

**Review by michael-mosler — APPROVED** [review]

**Pull request merged**